server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2114,8 +2115,9 @@ public void handlePostEntitlement(PoolManager manager, Consumer consumer,
                 }
             }
             List<Pool> subPoolsForStackIds = null;
-            // Share consumers should not contribute to the sharing org's stack
-            if (!stackIds.isEmpty() && !consumer.isShare()) {
+            // Manifest and Share consumers should not contribute to the sharing org's stack,
+            // as Share consumers should not have created a stack derived pool in the first place.
+            if (!stackIds.isEmpty() && !consumer.isShare() && !consumer.getType().isManifest()) {
```

Comment:
For consistency sake should we do `consumer.getType().isShare()` or `consumer.isManifest()`?